filters/united-states/hawaii/honolulu/public-art-json-to-json.js

```diff
@@ -1,47 +1,86 @@
-const filter = function (data, std_lib, stringify, skip_errors) {
+const filter = function (data, std_lib, schema, validator, stringify) {
+    // check for standard library and pull out required functions
+    if (!std_lib) {
+        throw "standard library not provided";
+    }
+    let add_required = std_lib.get("add_required");
+    let add_if_not_null = std_lib.get("add_if_not_null")
+    let remove_if_null = std_lib.get("remove_if_null");
+    let remove_if_empty = std_lib.get("remove_if_empty");
+    let validate_params = std_lib.get("validate_params");
+    let create_dates_template = std_lib.get("create_dates_template");
+    let remove_null_date_fields = std_lib.get("remove_null_date_fields");
+
+    // validate parameters object
+    schema = validate_params(schema, validator);
+
+    // convert JSON data to object form
     if (typeof data === 'string' || data instanceof String) {
         data = JSON.parse(data);
     }
 
+    // define new data and errors array
     let new_data = [];
+    let errors = [];
 
     data.map(d => {
         let item = {};
+        let skip = false;
 
-        item.name = d.title;
-        if (item.name === undefined) {
-            item.name = "";
-            console.log(`Data name not found for art with description:\n ${d.description}`);
+        // add name (required)
+        if (!add_required(item, "name", d, d.title, errors)) {
+            skip = true;
         }
 
-        let coordinates = { longitude: parseFloat(d.longitude), latitude: parseFloat(d.latitude) };
-        if (coordinates.longitude === undefined || coordinates.latitude === undefined) {
-            console.log(`Data coordinates not found for art with description:\n ${d.description}`);
+        // add coordinates (required)
+        let coordinates = {};
+        if (!add_required(coordinates, "longitude", d, parseFloat(d.longitude), errors)) {
+            skip = true;
+        }
+        if (!add_required(coordinates, "latitude", d, parseFloat(d.latitude), errors)) {
+            skip = true;
         }
         item.coordinates = coordinates;
 
-        let details = {
-            creator: d.creator,
-            credit: d.credit,
-            date: d.date,
-            description: d.description,
-            location: d.location,
-            imagefile: d.imagefile,
-            objectid: d.objectid,
-            discipline: d.discipline,
-            access: d.access
-        };
-        
-        item.details = details;
-
-        new_data.push(item);
+        item.dates = create_dates_template();
+        add_if_not_null(item.dates.installed, "year", d.date)
+        remove_null_date_fields(item)
+
+        add_if_not_null(item, "artist", d.creator);
+        add_if_not_null(item, "description", d.description);
+        add_if_not_null(item, "area", d.location);
+        add_if_not_null(item, "type", d.discipline);
+        item.image_urls = [add_if_not_null(d.imagefile)];
```

Comment:
this will add true or false to image_urls and not the actual url. Same suggestion as above